modules/twinklespeedy.js

```diff
@@ -1010,6 +1013,156 @@ Twinkle.speedy.initDialog = function twinklespeedyInitDialog(callbackfunc) {
+};
+
+Twinkle.speedy.instantDecline.main = function twinklespeedyInstantDeclineMain(pageObj) {
+	// finds the db template
+	const text = pageObj.getPageText();
+	const tag = /(?:\{\{\s*(db|delete|db-.*?|speedy deletion-.*?)(?:\s*\||\s*\}\}))/.exec(text)[1];
+	let wikitextObj = new Morebits.wikitext.page(text);
+
+	// and then removes it
+	wikitextObj.removeTemplate(tag);
+
+	if (wikitextObj.getText()[0] === '\n') {
+		wikitextObj = new Morebits.wikitext.page(wikitextObj.getText().slice(1));
+	}
+
+	pageObj.setPageText(wikitextObj.getText());
+
+	const splitTag = tag.split(/-(.*)/).length > 1 ? tag.split(/-(.*)/)[1] : tag;
+
+	if (splitTag in Twinkle.speedy.normalizeHash) {
+		if (Twinkle.speedy.normalizeHash[splitTag] === 'db') {
+				pageObj.setEditSummary('Declining speedy deletion (no code provided).');
+		} else {
+				pageObj.setEditSummary('Declining speedy deletion ([[WP:CSD#' + Twinkle.speedy.normalizeHash[splitTag].toUpperCase() + '|CSD ' + Twinkle.speedy.normalizeHash[splitTag].toUpperCase() + ']]).');
+		}
+	} else {
+		pageObj.setEditSummary('Declining speedy deletion (multiple criteria).');
+	}
+	pageObj.setChangeTags(Twinkle.changeTags);
+	pageObj.save(window.location.reload.bind(window.location));
+};
+
+Twinkle.speedy.createCodexButton = function twinkleSpeedyCreateCodexButton({ label, action = 'neutral', weight = 'normal', size = 'medium', onClick }) {
+	const button = document.createElement('button');
+	button.classList.add('cdx-button');
+
+	// Action classes
+	if (action !== 'neutral') {
+		button.classList.add(`cdx-button--action-${action}`);
+	}
+
+	// Weight classes
+	if (weight !== 'normal') {
+		button.classList.add(`cdx-button--weight-${weight}`);
+	}
+
+	// Size classes
+	if (size !== 'medium') {
+		button.classList.add(`cdx-button--size-${size}`);
+	}
+
+	button.textContent = label;
+
+	if (typeof onClick === 'function') {
+		button.addEventListener('click', onClick);
+	}
+
+	return button;
+};
+
+// Adds a quick "Delete" button to a speedy deletion template
+Twinkle.speedy.addButton = function twinklespeedyAddButton() {
+	if (!Morebits.userIsSysop) {
+		return;
+	}
+
+	const buttonNodes = {
+		delete: Twinkle.speedy.createCodexButton( { label: 'Delete', action: 'destructive', weight: 'primary', size: 'medium', onClick: Twinkle.speedy.instantDelete } ),
+		decline: Twinkle.speedy.createCodexButton( { label: 'Decline', action: 'neutral', weight: 'normal', size: 'medium', onClick: Twinkle.speedy.instantDecline } )
+	};
+
+	const buttonContainer = document.createElement('div');
+	buttonContainer.style.display = 'flex';
+	buttonContainer.style.gap = '1em';
+	buttonContainer.style.justifyContent = 'center';
+
+	// Append the buttons to the container
+	buttonContainer.appendChild(buttonNodes.delete);
+	buttonContainer.appendChild(buttonNodes.decline);
+
+	const contestButton = document.querySelector(
+		'table.ambox-speedy form[name="commentbox"].mw-inputbox-form-inline'
```

Comment:
Please add a class or id to the button and avoid such complex and fragile selectors